api/session.go

```diff
@@ -179,6 +180,8 @@ func (s *Session) RenewPeriodic(initialTTL string, id string, q *WriteOptions, d
 			// Attempt a session destroy
 			s.Destroy(id, q)
 			return nil
+		case <-ctx.Done():
```

Comment:
I wrote a test for this and realized that the canceled context in `q` will make this fail immediately. I think it's probably better to let the session expire on its own for this case since the workaround for that is pretty subtle and gross, and this is just a best effort remove. I've got a branch based on this PR so I'll make the change there.